### PR TITLE
Collect logs only for running containers

### LINKED ISSUE #688: flake: container "scylla" in pod "basic-us-east-1-us-east-1a-1" is terminated
https://github.com/scylladb/scylla-operator/runs/3030570314?check_suite_focus=true#step:11:150
```
• Failure in Spec Teardown (AfterEach) [265.622 seconds]
ScyllaCluster
github.com/scylladb/scylla-operator/test/e2e/set/scyllacluster/scyllacluster_scaling.go:16
  should support scaling [AfterEach]
  github.com/scylladb/scylla-operator/test/e2e/set/scyllacluster/scyllacluster_scaling.go:21

  Unexpected error:
      <*errors.StatusError | 0xc0008b4320>: {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "container \"scylla\" in pod \"basic-us-east-1-us-east-1a-1\" is terminated",
              Reason: "BadRequest",
              Details: nil,
              Code: 400,
          },
      }
      container "scylla" in pod "basic-us-east-1-us-east-1a-1" is terminated
  occurred

  github.com/scylladb/scylla-operator/test/e2e/framework/framework.go:302

  Full Stack Trace
  github.com/scylladb/scylla-operator/test/e2e/framework.(*Framework).afterEach(0xc0004fe780)
  	github.com/scylladb/scylla-operator/test/e2e/framework/framework.go:302 +0x585
  github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc00022ea80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
  	github.com/onsi/ginkgo@v1.15.0/internal/leafnodes/runner.go:113 +0xa3
  github.com/onsi/ginkgo/internal/leafnodes.(*runner).run(0xc00022ea80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
  	github.com/onsi/ginkgo@v1.15.0/internal/leafnodes/runner.go:64 +0x15c
  github.com/onsi/ginkgo/internal/leafnodes.(*SetupNode).Run(0xc00000e4f0, 0x1c1ba40, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
  	github.com/onsi/ginkgo@v1.15.0/internal/leafnodes/setup_nodes.go:15 +0x87
  github.com/onsi/ginkgo/internal/spec.(*Spec).runSample.func1(0xc000753270, 0xc0005d40f0, 0x1c1ba40, 0xc000303e80)
  	github.com/onsi/ginkgo@v1.15.0/internal/spec/spec.go:180 +0x3cd
  github.com/onsi/ginkgo/internal/spec.(*Spec).runSample(0xc0005d40f0, 0x0, 0x1c1ba40, 0xc000303e80)
  	github.com/onsi/ginkgo@v1.15.0/internal/spec/spec.go:218 +0x809
  github.com/onsi/ginkgo/internal/spec.(*Spec).Run(0xc0005d40f0, 0x1c1ba40, 0xc000303e80)
  	github.com/onsi/ginkgo@v1.15.0/internal/spec/spec.go:138 +0xf2
  github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpec(0xc000328160, 0xc0005d40f0, 0x2)
  	github.com/onsi/ginkgo@v1.15.0/internal/specrunner/spec_runner.go:200 +0x111
  github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).runSpecs(0xc000328160, 0x1)
  	github.com/onsi/ginkgo@v1.15.0/internal/specrunner/spec_runner.go:170 +0x147
  github.com/onsi/ginkgo/internal/specrunner.(*SpecRunner).Run(0xc000328160, 0xc00067f120)
  	github.com/onsi/ginkgo@v1.15.0/internal/specrunner/spec_runner.go:66 +0x117
  github.com/onsi/ginkgo/internal/suite.(*Suite).Run(0xc00038df10, 0x7fa429cce610, 0x27d1550, 0x1a38275, 0x19, 0xc00020a020, 0x2, 0x2, 0x1c54f58, 0xc000303e80, ...)
  	github.com/onsi/ginkgo@v1.15.0/internal/suite/suite.go:79 +0x546
  github.com/onsi/ginkgo.RunSpecsWithCustomReporters(0x1c1bac0, 0x27d1550, 0x1a38275, 0x19, 0xc00020a000, 0x2, 0x2, 0x2)
  	github.com/onsi/ginkgo@v1.15.0/ginkgo_dsl.go:229 +0x218
  github.com/onsi/ginkgo.RunSpecsWithDefaultAndCustomReporters(0x1c1bac0, 0x27d1550, 0x1a38275, 0x19, 0xc000554010, 0x1, 0x1, 0x1)
  	github.com/onsi/ginkgo@v1.15.0/ginkgo_dsl.go:217 +0xad
  github.com/scylladb/scylla-operator/pkg/cmd/tests.(*RunOptions).run(0xc0000d2800, 0x1c4db78, 0xc0004fe000, 0x1c1cbe0, 0xc0000b2000, 0x1c1cc00, 0xc0000b2008, 0x1c1cc00, 0xc0000b2010, 0x1, ...)
  	github.com/scylladb/scylla-operator/pkg/cmd/tests/tests_run.go:196 +0x257
  github.com/scylladb/scylla-operator/pkg/cmd/tests.(*RunOptions).Run(0xc0000d2800, 0x1c1cbe0, 0xc0000b2000, 0x1c1cc00, 0xc0000b2008, 0x1c1cc00, 0xc0000b2010, 0x1a1ca58, 0x3, 0x0, ...)
  	github.com/scylladb/scylla-operator/pkg/cmd/tests/tests_run.go:149 +0x357
  github.com/scylladb/scylla-operator/pkg/cmd/tests.NewRunCommand.func1(0xc00054e840, 0xc0003fe7a0, 0x0, 0x1, 0x0, 0x0)
  	github.com/scylladb/scylla-operator/pkg/cmd/tests/tests_run.go:81 +0xfd
  github.com/spf13/cobra.(*Command).execute(0xc00054e840, 0xc0003fe790, 0x1, 0x1, 0xc00054e840, 0xc0003fe790)
  	github.com/spf13/cobra@v1.1.1/command.go:850 +0x472
  github.com/spf13/cobra.(*Command).ExecuteC(0xc00054e580, 0xc0000b2000, 0x1c1cc00, 0xc0000b2008)
  	github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
  github.com/spf13/cobra.(*Command).Execute(...)
  	github.com/spf13/cobra@v1.1.1/command.go:895
  main.main()
  	github.com/scylladb/scylla-operator/cmd/scylla-operator-tests/scylla-operator-tests.go:37 +0x23d
```

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
When we scale down ScyllaCluster some containers may still be in a terminating state, which is fine because these are not accounted to the state of the ScyllaCluster. We'll skip log collection for non-running containers.

**Which issue is resolved by this Pull Request:**
Resolves #688

### CODE REVIEW

**Comment by tnozicka** [comment]
(clean rebase)

**Comment by tnozicka** [comment]
(clean rebase)

File: test/e2e/framework/dump.go [hunk 1]
```diff
@@ -46,58 +47,67 @@ func retrieveContainerLogs(ctx context.Context, podClient corev1client.PodInterf
...
-
-		hasPrevious := false
-		for _, cs := range pod.Status.ContainerStatuses {
-			if cs.Name != c.Name {
-				continue
-			}
-
-			if cs.RestartCount > 0 {
-				hasPrevious = true
+		if cs.State.Running != nil {
```
Comment (zimnx) [inline test/e2e/framework/dump.go:86]: shouldn't we also check the value of running?

Comment (tnozicka) [reply test/e2e/framework/dump.go:86]: I don't think we care about the value of the timestamp. Looks like kube checks similarly:
https://github.com/kubernetes/kubernetes/blob/cdf1b17bb8bcbf8a420d7e0a11336d09b0c37730/staging/src/k8s.io/kubectl/pkg/describe/describe.go#L1956-L1958

Comment (zimnx) [reply test/e2e/framework/dump.go:86]: oh i thought it's a `*bool`.

**Review by zimnx — APPROVED** [review]
LGTM

**Pull request merged**